packages/components/inputs/search-select-input/src/search-select-input.js

```diff
@@ -0,0 +1,215 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import { SafeHTMLElement } from '@commercetools-uikit/utils';
+import AsyncSelectInput from '@commercetools-uikit/async-select-input';
+import {
+  CustomSelectInputOption,
+  SearchIconDropdownIndicator,
+} from '@commercetools-uikit/select-utils';
+import messages from './messages';
+import { SearchSelectInputWrapper } from './search-select-input.styles';
+
+const SearchSelectInput = (props) => {
+  const intl = useIntl();
+  const noOptionsMessage =
+    props.noOptionsMessage ||
+    (() => intl.formatMessage(messages.noOptionsMessage));
+  const loadingMessage =
+    props.loadingMessage || intl.formatMessage(messages.loadingOptionsMessage);
+  const placeholder =
+    props.placeholder || intl.formatMessage(messages.placeholderMessage);
+
+  // eslint-disable-next-line react/display-name
+  const OptionComponent = (optionInnerProps) => (
+    <CustomSelectInputOption
+      optionType={props.optionType}
+      optionInnerProps={optionInnerProps}
+    />
+  );
+
+  return (
+    <SearchSelectInputWrapper>
+      <AsyncSelectInput
+        {...props}
+        components={{
```

Comment:
Nit:

```js
const components = React.useMemo(() => ({
   Option: OptionComponent,
   ...props.component,
   DropdownIndicator: SearchIconDropdownIndicator
}), [props.component, OptionComponent, SearchIconDropdownIndicator])
```

under the hope that a static reference helps the underlying AsyncSelectInput.

Reply:
will add these changes. however it will also either wrapping the `OptionComponent` in its callback (since it is used inside memo and it has `optionType` dependency) or moving it to the `useMemo`

Reply:
https://github.com/commercetools/ui-kit/pull/1564/commits/3da8c419a6a663bd32880a5559e9685ace86751a